test/features/interceptor/authorization_interceptor_test.dart

```diff
@@ -0,0 +1,247 @@
+          newConfig: OIDCFixtures.oidcConfiguration);
+
+        final result = authorizationInterceptors.validateToRefreshToken(responseStatusCode: responseStatusCode);
+
+        expect(result, true);
+      });
+
+      test('validateToRefreshToken should return false when condition `responseStatusCode == 500`', () async {
+        const responseStatusCode = 500;
+
+        authorizationInterceptors.setTokenAndAuthorityOidc(
+          newToken: OIDCFixtures.tokenOidcExpired,
+          newConfig: OIDCFixtures.oidcConfiguration);
+
+        final result = authorizationInterceptors.validateToRefreshToken(responseStatusCode: responseStatusCode);
+
+        expect(result, false);
+      });
+
+      test('validateToRefreshToken should return false when condition `OidcConfiguration is null`', () async {
+        const responseStatusCode = 401;
+
+        authorizationInterceptors.setTokenAndAuthorityOidc(
+          newToken: OIDCFixtures.tokenOidcExpired,
+          newConfig: null);
+
+        final result = authorizationInterceptors.validateToRefreshToken(responseStatusCode: responseStatusCode);
+
+        expect(result, false);
+      });
+
+      test('validateToRefreshToken should return false when condition `token is empty`', () async {
+        const responseStatusCode = 401;
+
+        authorizationInterceptors.setTokenAndAuthorityOidc(
+          newToken: OIDCFixtures.tokenOidcWithTokenEmpty,
+          newConfig: OIDCFixtures.oidcConfiguration);
+
+        final result = authorizationInterceptors.validateToRefreshToken(responseStatusCode: responseStatusCode);
+
+        expect(result, false);
+      });
+
+      test('validateToRefreshToken should return false when condition `refreshToken is empty`', () async {
+        const responseStatusCode = 401;
+
+        authorizationInterceptors.setTokenAndAuthorityOidc(
+          newToken: OIDCFixtures.tokenOidcWithRefreshTokenEmpty,
+          newConfig: OIDCFixtures.oidcConfiguration);
+
+        final result = authorizationInterceptors.validateToRefreshToken(responseStatusCode: responseStatusCode);
+
+        expect(result, false);
+      });
+
+      test('validateToRefreshToken should return false when condition `Time not expired`', () async {
+        const responseStatusCode = 401;
+
+        final tokenOidc = TokenOIDC(
+          'dab123',
+          TokenId('dab123'),
+          'dab456',
+          expiredTime: DateTime.now().add(const Duration(days: 1))
+        );
+
+        authorizationInterceptors.setTokenAndAuthorityOidc(
+          newToken: tokenOidc,
+          newConfig: OIDCFixtures.oidcConfiguration);
+
+        final result = authorizationInterceptors.validateToRefreshToken(responseStatusCode: responseStatusCode);
+
+        expect(result, false);
+      });
+    });
+
+    group('QueuedInterceptorsWrapper test', () {
+      test('The request not should is not blocked when returning onError in case the token expires', () async {
```

Comment:
`The request should not be blocked when returning onError in case the token expires`